### PR TITLE
Waffle Iron - A syndicate uplink laser pistol

### PR DESCRIPTION
**Pull request by ProphetCrow**
## About the PR
Adds the "Waffle Iron", which is a syndicate energy pistol that is based off the svalinn laser pistol (SLP). The weapon is currently as powerful as the laser rifle (LR), and starts with a medium powercell, giving it two less shots than the LR. Currently, this weapon is able to be purchased in any uplink for 8TC. 

## Why / Balance
It's cool. / It gives syndicates a laser weapon that operates uniquely compared to all the other energy weapons in the game, barring the SLP. 

## Technical details
*Added the WeaponLaserWaffle entity to the battery guns prototype. The code is similar to the SLP

*Added  the following component: 
- type: HitscanBatteryAmmoProvider
    proto: RedLaser

To both the Svalinn Laser Pistol and Waffle Iron during testing to see if I can change the weapon damage without changing the battery "magazine" damage. This added the "Inspect damage" verb to the weapons, but was bugged, displaying incorrect damage values as the damage is based off the magazine loaded and not the gun itself. Left the component attached, as without it, you cannot inspect damage values. 

*Adjusted projectile "RedLightLaser" damage from 7 to 14, this affects the Svalinn Laser Pistol and the Waffle Iron, doubling its base damage. This is a bandaid fix to the previously mentioned bug. 

*Added item to the weaponry tab of the uplink with its own description

*Known bug - The SLP and the Waffle Iron's ammo counter displays "x00" regardless of remaining shots. However, inspecting the weapon will accurately display the remaining charges of the tool. This is likely due to the ammo capacity being dynamically based off of powercell size instead of traditional energy tanks.  

*Known bug - Inserting a microreactor cell into the SLP or the Waffle Iron will not update the graphic whenever it gains new charges. If you deplete the charge of a microreactor cell in either weapon and let it recharge, it's sprite will display empty until the battery is ejected and reinserted. 

## Media

https://cdn.discordapp.com/attachments/1010560161888350340/1260071952534732820/Delta-v_2024-07-08_23-02-42.mp4?ex=668dfcf4&is=668cab74&hm=0b0b691c6c1fd0e8f3bc4cf66c82d4272149ca5e3cccbb9c1c719dd560516f43& / 

https://discord.com/channels/968983104247185448/1010560161888350340/1260071954694803518

- [ X ] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Other

I plan to make this weapon reverse engineer-able in the future with a blue-kie sprite variant. As this affects balance, this will be in a separate PR. 

**Changelog**

:cl: Crow
- add: Added Waffle Iron - A syndicate uplink laser pistol
- tweak: Adjusted damage of the power cell based energy weapons
- fix: Added inspection verb to the Svalinn Laser Pistol

### CODE REVIEW

**Comment by ProphetCrow** [comment]
> This pull request has conflicts, please resolve those before we can evaluate the pull request.

resolved conflicts

**Comment by Sifb26** [comment]
I feel like the damage should be nerfed a tad since it’s a little bit nutty that an epi antagonist can just get the upgraded version of the antique laser pistol since they can just make microreactors or print 50 high caps. I’d think that 12 damage would be a fair nerf since it’s just so easy to collect power cells

**Comment by ProphetCrow** [comment]
> I feel like the damage should be nerfed a tad since it’s a little bit nutty that an epi antagonist can just get the upgraded version of the antique laser pistol since they can just make microreactors or print 50 high caps. I’d think that 12 damage would be a fair nerf since it’s just so easy to collect power cells

Sure, if the maintainers want to nerf the damage values, the code is in Resources/Prototypes/Entities/Objects/Weapons/Guns/Projectiles/hitscan.yml , RedLightLaser line. I modeled the damage after the laser rifle since I think that weapon is overall mediocre, but potentially infinite ammo is a concern. 

Telecrystal cost is in Resources/Prototypes/DeltaV/Catalog/uplink_catalog.yml if they want to adjust that too, I think 8 is fair when compared to the energysword and python. 

It should also be noted that the Antique laser uses RedMediumLaser - which does 17 damage currently.

**Comment by BlitzTheSquishy** [comment]
There is somewhat a balancing reason on why syndicates arent provided energy weapons (disregarding the CHIMP upgrade chip and emag for the protolathe)

May want it to be somewhat explicitly weaker than the laser rifle

File: Resources/Prototypes/Entities/Objects/Weapons/Guns/Projectiles/hitscan.yml [hunk 1]
```diff
@@ -65,7 +65,7 @@
   id: RedLightLaser
   damage:
     types:
-      Heat: 7
+      Heat: 14 #7
```
Comment (DangerRevolution) [inline Resources/Prototypes/Entities/Objects/Weapons/Guns/Projectiles/hitscan.yml:68]:
```suggestion
      Heat: 14 # Delta-V : upgrades damage from 7 to 14
```

File: Resources/Prototypes/Entities/Objects/Weapons/Guns/Battery/battery_guns.yml
Comment (DangerRevolution) [inline Resources/Prototypes/Entities/Objects/Weapons/Guns/Battery/battery_guns.yml:1]: Delta-V folder please.

**Review by DangerRevolution — CHANGES_REQUESTED** [review]

**Comment by TadJohnson00** [comment]
As Blitz stated, there are pre-existing standards for balance regarding traitors. I'm generally opposed to granting them a laser weapon, first of all, let alone one with significant damage and the capability to be refilled from battery cells.

I advise against merging this PR until a proper discussion can be had on weapons balance and where traitors fall with regards to that.

**Comment by TadJohnson00** [comment]
@ProphetCrow Are you still working on this?

**Comment by PubliclyExecutedPig** [comment]
ok but would EMP's affect it

**Comment by Avalon-Proto** [comment]
Yes because EMPs effect all power cells

**Comment by TadJohnson00** [comment]
Closing this PR due to inactivity.

**Pull request closed without merging**